* Not using configure, but just thinking you've got it right means you
  break, because you're not right.  round() is available on Solaris

--- psycopg/config.h
+++ psycopg/config.h
@@ -102,7 +102,7 @@
 #define inline
 #endif
 
-#if (defined(_WIN32) && !defined(__GNUC__)) || defined(__sun__) || defined(sun)
+#if (defined(_WIN32) && !defined(__GNUC__))
 /* what's this, we have no round function either? */
 static double round(double num)
 {
